File: Stronghold/Stronghold.Application/DTOs/Response/UserReviewResponse.cs
namespace Stronghold.Application.DTOs.Response
{
    public class UserReviewResponse
    {
        public int Id { get; set; }
        public string SupplementName { get; set; } = string.Empty;
        public int Rating { get; set; }
        public string? Comment { get; set; }
    }
}

File: packages/stronghold_core/lib/models/responses/user_review_response.dart
/// User-facing review response (member's own reviews)
class UserReviewResponse {
  final int id;
  final String supplementName;
  final int rating;
  final String? comment;

  const UserReviewResponse({
    required this.id,
    required this.supplementName,
    required this.rating,
    this.comment,
  });

  factory UserReviewResponse.fromJson(Map<String, dynamic> json) {
    return UserReviewResponse(
      id: json['id'] as int,
      supplementName: json['supplementName'] as String,
      rating: json['rating'] as int,
      comment: json['comment'] as String?,
    );
  }
}
